R/lightgbm.R

```diff
@@ -185,10 +182,49 @@ process_parallelism <- function(args) {
   args
 }
 
+process_data <- function(args, x, y, validation, missing_validation,
+                         early_stopping_rounds) {
+  n <- nrow(x)
+
+  needs_validation <- !missing_validation || !is.null(early_stopping_rounds)
+
+  if (!needs_validation) {
+    trn_index <- 1:n
+    val_index <- NULL
+  } else {
+    m <- min(floor(n * (1 - validation)) + 1, n - 1)
```

Comment:
Should we message in the case that a user supplies `stop_iter` (`early_stopping_rounds`) but doesn't supply `validation` (`valids`) to let them know we're going to set aside a small proportion of the training data for validation and point them to the `validation` argument?

Reply:
no, we don't issue a warning for that.